apollo-client/src/main/java/com/ctrip/framework/apollo/AutoConfigChangeListener.java

```diff
@@ -0,0 +1,17 @@
+package com.ctrip.framework.apollo;
+
+        import com.ctrip.framework.apollo.model.ConfigChangeEvent;
+
+/**
+ * @author tony Jiang(258737400@qq.com)
+ */
+public interface AutoConfigChangeListener {
```

Comment:
I don't think we need a new list of AutoConfigChangeListeners. If we want to make sure some listener is executed first, we could create a a new interface such as `PriorityOrderedConfigChangeListener` that extends ConfigChangeListener and allow the user to provide its order?

Reply:
My idea is to provide a flexible refresh mode for users:

1. based on  ApolloConfigChangeListener annotation, let users write their own refresh code

2. based on AutoConfigChangeListener annotation, automatic assembly of data for simple data types

In addition, with regard to #898, I do not agree with this coercion, which forces all configuration variables to be refreshed automatically, thus losing the flexibility of the user.

Above all, I think it's better to provide an independent <AutoConfigChangeListener> annotation, because this is the two separate listener behavior, and there is no distinction between execution sequence.